aten/src/ATen/native/sparse/SparseBinaryOpIntersectionCommon.h

```diff
@@ -264,6 +264,11 @@ void _sparse_binary_op_intersection_kernel_impl(
 
     auto hash = at::empty({probably_coalesced._nnz()}, indices.options().dtype(kLong));
 
+    if (indices.numel() == 0) {
+      hash.zero_();
+      return hash;
+    }
```

Comment:
No, this is logically incorrect...

Reply:
Can you please elaborate what do you mean here? I'm returning as soon as size of `hash` is computed (on the previous line) 
To be frank, I have a hard time following what this code is doing, and only trying to solve UBSan violation. One trivial way of doing that would be to add ternary on line 284
```
    const auto* RESTRICT ptr_indices_dim = ptr_indicies ? ptr_indices + nnz_idx * indices_nnz_stride : nullptr;
```

But it feels like it would return hash full of zeroes anyway, so it's reasonable to do it here, before iterator is constructed and kernel is launched. Further optimizations, imo, belong to separate PR (but yes, I agree with you, it's possible to return even earlier)

Reply:
Could you please use a ternary for now, and I could follow-up with a PR next? I agree, the kernel is rather non-trivial, and having `hash == 0` is logically incorrect since values with equal hash values do belong to the same equivalence class in the intersection.

Reply:
Sure, I can, but that what it would return, wouldn't it? (i.e. would you be more comfortable, if I'll replace `numel() == 0` with `sparse_dim == 0`?

Reply:
Yes, testing against `sparse_dim` is better because it should imply zero nnz, and we can return `hash` as is. Unfortunately, `numel == 0` does not imply `nnz == 0`, one of dimensions of `hash`.

Reply:
Ok, update, will create TODO issues later